new-integration-tests/src/test/java/oracle/weblogic/kubernetes/ItSimpleTraefikValidation.java

```diff
@@ -0,0 +1,209 @@
+
+    assertThat(helmList(tfHelmParams))
+        .as("Test helmList returns true")
+        .withFailMessage("helmList() did not return true")
+        .isTrue();
+
+    // waiting for an async operation to complete.
+    // we first wait 30 seconds, then we check if the traefik pod is running.
+    with().pollDelay(30, SECONDS)
+        // we check again every 10 seconds.
+        .and().with().pollInterval(10, SECONDS)
+        // this listener lets us report some status with each poll
+        .conditionEvaluationListener(
+            condition -> logger.info(
+                "Waiting for traefik to be running (elapsed time {0}ms, remaining time {1}ms)",
+                condition.getElapsedTimeInMS(),
+                condition.getRemainingTimeInMS()))
+        // and here we can set the maximum time we are prepared to wait
+        .await().atMost(5, MINUTES)
+        .until(traefikIsRunning(tfNamespace));
+  }
+
+  @Test
+  @Order(2)
+  @DisplayName("Upgrade the Traefik release")
+  public void testUpgradingTraefik() {
+
+    // create a new unique namespace for wls domain
+    domainNamespace1 = createNamespace();
+    logger.info(String.format("Created a new namespace called %s", domainNamespace1));
+
+    // helm install parameters
+    tfHelmParams = new HelmParams()
+            .releaseName("traefik-operator")
+            .namespace(tfNamespace)
+            .chartDir("stable/traefik");
+
+    TraefikParams tfParams = new TraefikParams()
+            .helmParams(tfHelmParams)
+            .nameSpaces("{" + tfNamespace + "," + domainNamespace1 + "}");
+
+    // upgrade Traefik
+    assertThat(upgradeTraefik(tfParams))
+            .as("Test upgradeTraefik returns true")
+            .withFailMessage("upgradeTraefik() did not return true")
+            .isTrue();
+    logger.info(String.format("Traefik is upgraded in namespace %s", tfNamespace));
+
+    assertThat(helmGetExpectedValues(tfHelmParams, domainNamespace1))
+            .as("Test helmGetExpectedValues returns true")
+            .withFailMessage("helmGetExpectedValues() did not return true")
+            .isTrue();
+  }
+
+  @Test
+  @Order(3)
+  @DisplayName("Create a ingress per domain")
+  public void testCreatingIngress() {
+    // helm install parameters for ingress
+    ingressParam = new HelmParams()
+            .releaseName("sample-domain1-ingress")
+            .namespace(domainNamespace1)
+            .chartDir("../kubernetes/samples/charts/ingress-per-domain");
+
+    assertThat(createIngress(ingressParam, "sample-domain1", "sample-domain1.org"))
+            .as("Test createIngress returns true")
+            .withFailMessage("createIngress() did not return true")
+            .isTrue();
+    logger.info(String.format("ingress is upgraded in namespace %s", domainNamespace1));
+
+  }
+
+  @AfterAll
+  public void tearDown() {
+
+    // uninstall ingress
+    if (ingressParam != null) {
```

Comment:
I believe we agreed that @sankarpn 's test watcher would clean-up the test namespaces after he collects diagnostics.  So, we don't want to delete resources or even uninstall Helm releases.

Reply:
I will leave the tearDown() method for now. Once Sankar's PR is in, I can remove it.

Reply:
ok, but please add a `// TODO ` so we don't forget

Reply:
done